test/Altinn.Profile.Tests/IntegrationTests/API/Controllers/DashboardNotificationAddressesControllerTests.cs

```diff
@@ -0,0 +1,301 @@
+                .ReturnsAsync(parentOrg);
+            
+            _factory.OrganizationNotificationAddressRepositoryMock
+                .Setup(r => r.GetOrganizationAsync(parentOrg, It.IsAny<CancellationToken>()))
+                .ReturnsAsync(parentUnit);
+            
+            HttpClient client = _factory.CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, $"/profile/api/v1/dashboard/organizations/{requestedOrg}/notificationaddresses");          
+            request = CreateAuthorizedRequestWithScope(request);
+
+            // Act
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            string responseContent = await response.Content.ReadAsStringAsync();
+            var actual = JsonSerializer.Deserialize<List<DashboardNotificationAddressResponse>>(responseContent, _serializerOptions);
+
+            Assert.NotNull(actual);
+
+            var expectedAddresses = parentUnit.NotificationAddresses?
+                .Where(n => n.IsSoftDeleted != true && n.HasRegistryAccepted != false)
+                .ToList() ?? new List<NotificationAddress>();
+
+            Assert.Equal(expectedAddresses.Count, actual.Count);
+            
+            Assert.All(actual, a => Assert.Equal(requestedOrg, a.RequestedOrgNumber));
+            Assert.All(actual, a => Assert.Equal(parentOrg, a.SourceOrgNumber));
+        }
+
+        [Fact]
+        public async Task GetAllNotificationAddressesForAnOrg_WhenNoMatchingOrganization_ReturnsNotFound()
+        {
+            // Arrange
+            string orgNumber = "error-org";            
+
+            _factory.OrganizationNotificationAddressRepositoryMock
+                .Setup(r => r.GetOrganizationsAsync(It.IsAny<List<string>>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(Enumerable.Empty<Organization>());
+
+            HttpClient client = _factory.CreateClient();
+
+            HttpRequestMessage httpRequestMessage = new(HttpMethod.Get, $"/profile/api/v1/dashboard/organizations/{orgNumber}/notificationaddresses");
+            httpRequestMessage = CreateAuthorizedRequestWithScope(httpRequestMessage);
+
+            // Act            
+            HttpResponseMessage response = await client.SendAsync(httpRequestMessage);          
+
+            // Assert
+            Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task GetAllNotificationAddressesForAnOrg_WhenNoAccess_ReturnsForbidden()
+        {
+            // Arrange
+            string orgNumber = "111111111";
+                                  
+            _factory.OrganizationNotificationAddressRepositoryMock
+                .Setup(r => r.GetOrganizationsAsync(It.IsAny<List<string>>(), It.IsAny<CancellationToken>()))
```

Comment:
This mock is never reached as the request is rejected at api level. Can be removed.